crates/net/network/src/transactions/fetcher.rs

```diff
@@ -33,27 +37,32 @@ use super::{
 /// new requests on announced hashes.
 #[derive(Debug)]
 #[pin_project]
-pub(crate) struct TransactionFetcher {
+pub struct TransactionFetcher {
     /// All peers with to which a [`GetPooledTransactions`] request is inflight.
-    pub(super) active_peers: LruMap<PeerId, u8, ByLength>,
+    pub active_peers: LruMap<PeerId, u8, ByLength>,
     /// All currently active [`GetPooledTransactions`] requests.
     ///
     /// The set of hashes encompassed by these requests are a subset of all hashes in the fetcher.
     /// It's disjoint from the set of hashes which are awaiting an idle fallback peer in order to
     /// be fetched.
     #[pin]
-    pub(super) inflight_requests: FuturesUnordered<GetPooledTxRequestFut>,
+    pub inflight_requests: FuturesUnordered<GetPooledTxRequestFut>,
     /// Hashes that are awaiting an idle fallback peer so they can be fetched.
     ///
     /// This is a subset of all hashes in the fetcher, and is disjoint from the set of hashes for
     /// which a [`GetPooledTransactions`] request is inflight.
-    pub(super) hashes_pending_fetch: LruCache<TxHash>,
+    pub hashes_pending_fetch: LruCache<TxHash>,
     /// Tracks all hashes in the transaction fetcher.
-    pub(super) hashes_fetch_inflight_and_pending_fetch: LruMap<TxHash, TxFetchMetadata, Unlimited>,
+    pub hashes_fetch_inflight_and_pending_fetch: LruMap<TxHash, TxFetchMetadata, Unlimited>,
     /// Filter for valid eth68 announcements.
-    pub(super) filter_valid_hashes: AnnouncementFilter,
+    pub filter_valid_hashes: AnnouncementFilter,
     /// Info on capacity of the transaction fetcher.
-    pub(super) info: TransactionFetcherInfo,
+    pub info: TransactionFetcherInfo,
+    /// [`FetchEvent`]s as a result of advancing inflight requests
+    #[pin]
+    pub fetch_events_head: UnboundedMeteredReceiver<FetchEvent>,
+    /// Handle for queueing [`FetchEvent`]s as a result of advancing inflight requests.
+    pub fetch_events_tail: UnboundedMeteredSender<FetchEvent>,
```

Comment:
what's this event channel for?

Reply:
intermediary storage of events

Reply:
it's a light weight queue-stack (sounds like an oxymoron). fifo structure, that can't be traversed, just popped and pushed to.